src/Security/EventListener/LastVisitedPageOnGetListener.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace AppBundle\Security\EventListener;
+
+use Symfony\Bundle\SecurityBundle\Security\FirewallMap;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class LastVisitedPageOnGetListener implements EventSubscriberInterface
+{
+    use TargetPathTrait;
+
+    private const FIREWALL_NAME = 'main';
+    private const IGNORED_ROUTES = [
+        'app_user_login',
+        'logout',
+        'asset_url',
+        'app_membership_activate',
+    ];
+
+    private $security;
+    private $firewallMap;
+    private $apiPathPrefix;
+
+    public function __construct(string $apiPathPrefix, Security $security, FirewallMap $firewallMap)
+    {
+        $this->apiPathPrefix = $apiPathPrefix;
+        $this->security = $security;
+        $this->firewallMap = $firewallMap;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::RESPONSE => ['onKernelResponse', -1],
+        ];
+    }
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $token = $this->security->getToken();
+
+        // Stop if not MasterRequest or not Response != 200
```

Comment:
or Response != 200 ?

Reply:
Yes, we will save the page as last visited if the response was succeeded, that's why we ignore responses with the status code different from 200.

Reply:
need remove `not` from comment :D

Reply:
ah :D